firefox-settings/pure.js

```diff
@@ -155,3 +155,37 @@ pref("browser.safebrowsing.provider.mozilla.pver", 0, locked);
+pref("browser.uitour.enabled", false); // https://trac.torproject.org/projects/tor/ticket/19047#comment:3
+
+pref("content.notify.interval", 100000); // page reflow timer, lower redrawn rendering timer, increases responsiveness but increase total load time
+pref("dom.security.https_only_mode", true, locked); // only allow https in all windows, including private browsing
+pref("network.auth.subresource-http-auth-allow", 1, locked); // block HTTP authentication credential dialogs
+pref("network.cookie.sameSite.noneRequiresSecure", true); // https://hacks.mozilla.org/2020/08/changes-to-samesite-cookie-behavior/
+pref("network.dns.disablePrefetch", true); // https://developer.mozilla.org/docs/Web/HTTP/Headers/X-DNS-Prefetch-Control
+pref("network.dns.disablePrefetchFromHTTPS", true); // https://developer.mozilla.org/docs/Web/HTTP/Headers/X-DNS-Prefetch-Control
+
+pref("network.dnsCacheExpiration", 3600); // Time DNS entries are cached in seconds.
+pref("network.http.max-connections", 1800); //Maximum number of any type of connections per server
+pref("network.http.max-persistent-connections-per-server", 10); //Maximum number of keep-alive type connections per server
+pref("network.http.max-urgent-start-excessive-connections-per-host", 5); //Number of connections that we can open beyond the standard parallelism limit defined by max-persistent-connections-per-server/-proxy to handle urgent-start marked requests
+pref("network.http.pacing.requests.enabled", false); //Disable pacing requests
+
+pref("network.predictor.enable-prefetch", false) // Disable speculative website loading
+pref("network.predictor.enabled", false); // Disable speculative website loading
+pref("network.prefetch-next", false); // https://developer.mozilla.org/en-US/docs/Glossary/Prefetch
+pref("network.ssl_tokens_cache_capacity", 10240); // Increase TLS token caching
+pref("nglayout.initialpaint.delay", 0); // How long firefox will wait before rendering page
+pref("nglayout.initialpaint.delay_in_oopif", 0); // How long firefox will wait before rendering page
+pref("privacy.globalprivacycontrol.enabled", true); // https://support.mozilla.org/1/firefox/126.0/Linux/en-US/global-privacy-control
+
+pref("privacy.query_stripping.strip_list", "__hsfp __hssc __hstc __s _hsenc _openstat dclid fbclid gbraid gclid hsCtaTracking igshid mc_eid ml_subscriber ml_subscriber_hash msclkid oft_c oft_ck oft_d oft_id oft_ids oft_k oft_lk oft_sk oly_anon_id oly_enc_id rb_clickid s_cid twclid vero_conv vero_id wbraid wickedid yclid"); // https://groups.google.com/a/mozilla.org/g/dev-platform/c/1vOSas0ptVQ?pli=1
+pref("urlclassifier.features.socialtracking.skipURLs", "*.instagram.com, *.twitter.com, *.twimg.com"); // allow embedded social content
+pref("urlclassifier.trackingSkipURLs", "*.reddit.com, *.twitter.com, *.twimg.com"); // allow embedded social content
+
+pref("widget.use-xdg-desktop-portal.file-picker", 1) // Use desktop portals' file picker
```

Comment:
I think it requires adding ``xdg-desktop-portal-impl`` or ``xdg-desktop-portal-gtk`` to the dependencies?

Reply:
The issue is that not every implementation of xdg-desktop-portal contains a file picker, which may cause dialogs to not work in Firefox. But I think this is a edge case.